APCAppCore/APCAppCore/Library/Categories/NewDataCollector/Data Sink/APCPassiveDataSink.m

```diff
@@ -0,0 +1,472 @@
+    
+    [self.healthKitCollectorQueue addOperationWithBlock:^{
+    
+        __typeof(self) strongSelf = weakSelf;
+        
+        NSString *stringToWrite = [self transformCollectorData:quantitySample];
+        
+        [APCPassiveDataSink createOrAppendString:stringToWrite
+                                          toFile:[strongSelf.folder stringByAppendingPathComponent:kCSVFilename]];
+        
+        [strongSelf checkIfDataNeedsToBeFlushed];
+    }];
+}
+
+- (NSString*)transformCollectorData:(id)dataSample
+{
+    return self.transformer(dataSample);
+}
+
+- (instancetype)initWithIdentifier:(NSString*)identifier columnNames:(NSArray*)columnNames operationQueueName:(NSString*)operationQueueName andDataProcessor:(CSVSerializer)transformer
+{
+    self = [super init];
+    
+    if (self)
+    {
+        //Unique configuration for collector
+        _identifier     = identifier;
+        _columnNames    = columnNames;
+        _transformer    = transformer;
+
+        if (!self.healthKitCollectorQueue) {
+            self.healthKitCollectorQueue = [NSOperationQueue sequentialOperationQueueWithName:operationQueueName];
+        }
+        
+        [self checkIfCSVStructureHasChanged];
+        
+        //General configuration for file management
+        NSString* documentsDir  = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES)[0];
+
+        _collectorsPath         = [documentsDir stringByAppendingPathComponent:kCollectorFolder];
+        
+        [APCPassiveDataSink createFolderIfDoesntExist:_collectorsPath];
+        [APCPassiveDataSink createFolderIfDoesntExist:[_collectorsPath stringByAppendingPathComponent:kUploadFolder]];
+        
+        [self loadOrCreateDataFiles];
+    }
+    
+    return self;
+}
+
+- (NSString*)collectorsUploadPath
+{
+    return [self.collectorsPath stringByAppendingPathComponent:kUploadFolder];
+}
+
+/*********************************************************************************/
+#pragma mark - Adding tracker
+/*********************************************************************************/
+
+- (void) loadOrCreateDataFiles
+{
+    self.folder = [self.collectorsPath stringByAppendingPathComponent:self.identifier];
+    
+    NSString*       infoFilePath    = [self.folder stringByAppendingPathComponent:kInfoFilename];
+    NSDictionary*   infoDictionary  = nil;
+
+    //Create log files
+    if (![[NSFileManager defaultManager] fileExistsAtPath:self.folder])
+    {
+        NSError* folderCreationError = nil;
+        
+        if (![[NSFileManager defaultManager] createDirectoryAtPath:self.folder
+                                       withIntermediateDirectories:YES
+                                                        attributes:@{
+                                                                     NSFileProtectionKey :
+                                                                         NSFileProtectionCompleteUntilFirstUserAuthentication
```

Comment:
Same file protection comment applies here, although not obvious that you want to be applying your file protection settings at the directory level.